common/unsafe/src/main/java/org/apache/spark/unsafe/types/ByteArray.java

```diff
@@ -101,4 +101,117 @@ public static long getPrefix(byte[] bytes) {
     }
     return result;
   }
+
+  /**
+   * Trims bytes representing spaces from both ends of this byte array.
+   *
+   * @param srcArr the source byte array
+   * @return this byte array with no bytes representing spaces at the start or end
+   */
+  public static byte[] trim(byte[] srcArr) {
+    return trim(srcArr, " ".getBytes());
```

Comment:
This could theoretically be locale-dependent. Why not just write in the byte 32 to be clear?

Reply:
Good idea.